### PR TITLE
Dual axis and dual extruder support for RUMBA board.

### PR DESCRIPTION
**Pull request by d00616**
This adds dual Y Axis support in an dual extruder configuration to the RUMBA board.

### CODE REVIEW

**Comment by d00616** [comment]
@boelle Builds perfectly without modifications. Thanks for closing my previous PR

**Comment by boelle** [comment]
yep this one is perfect, thou we are in bug fixing phase so it will not be merged until bug fixing is over

**Comment by Harpye** [comment]
Sorry about hijacking this  - but perhaps you could help?!?

Trying to use the standard Marlin 1.1 package on  RUMBA board ( Board 80)  I recognized the following!
using X;Y;Z and E0+E1 works fine, but additionally using E2 causes missbehaviour

I built a Gcode skeleton which works fine on the  E2 but after including some movement blocks of the other stages ( which also work well in standing alone usage) in between the existing E2/ T2 movements,  E2 does not work properly anymore .

I put a gcode comparison  in here:

http://forums.reprap.org/read.php?146,491725

( sorry but I'm brand new to that programming thingy and tired for rather a week now, to understand the  Marlin code ! I got stuck...)

Any help would absolutely be appreciated!

Hansjoerg

**Comment by d00616** [comment]
Sorry i cannot help here. At the moment i use this board for milling
only. I need E2 for an second Y axis motor to handle the weight of my X
axis. Do you have opened an issue?
https://github.com/MarlinFirmware/Marlin/issues

**Comment by Harpye** [comment]
I'll open one ! - thank you  for the hint!

I also don't use this board for printing  but need 6 channels ( better would be they could all be interpolated)!

**Comment by boelle** [comment]
do you still want this and is it ready?

**Comment by d00616** [comment]
Yes.

Am 07.06.2015 um 02:11 schrieb Bo Herrmannsen:

**Comment by d00616** [comment]
@Wackerbarth This was my third and last try to add correct RUMBA Support to Marlin.

Don't reopen this PR. I'm not interested wasting more of my time by contributing to Marlin.

**Comment by thinkyhead** [comment]
@d00616 Thanks for all your hard work, I personally appreciate it! We've just had a lot of balls in the air trying to get the next release ready, and as you know this PR has been "on-hold during [this lengthy] bugfix" round. I'll be happy to re-roll this for you and merge it soon. It has to be moved over to "MarlinDev" to fit with the new workflow, but if possible I will preserve your commits so you get the credit in the commit log.

**Comment by d00616** [comment]
Thanks. You don't need preserve my commits.

**Comment by d00616** [comment]
http://suchen.mobile.de/fahrzeuge/details.html?id=213387027&fuels=DIESEL&ambitCountry=DE&transmissions=AUTOMATIC_GEAR&sortOption.sortBy=searchNetGrossPrice&climatisation=MANUAL_OR_AUTOMATIC_CLIMATISATION&scopeId=C&damageUnrepaired=NO_DAMAGE_UNREPAIRED&sortOption.sortOrder=ASCENDING&categories=Van&features=ELECTRIC_WINDOWS&features=ELECTRIC_EXTERIOR_MIRRORS&features=AUXILIARY_HEATING&zipcode=14469&isSearchRequest=true&export=NO_EXPORT&zipcodeRadius=200&maxMileage=20000&minFirstRegistrationDate=2014-01-01&pageNumber=1

**Comment by thinkyhead** [comment]
@d00616 Interesting link! I'll be merging these changes shortly...

**Comment by thinkyhead** [comment]
@d00616 So, it appears that most of your PR was already incorporated into the code-base — at least, Dual Y support was already implemented. However the pins definitions for Dual Y weren't being overridden in the same manner as Dual Z were, which would have obviated the need to do anything special for RUMBA (or any other board).

In any case, I'm not too fond of the way that the pins were being overridden in `Configuration_adv.h` so I'm removing those lines and moving the standard definitions of the Y2 and Z2 pins over to `pins.h` where they can apply to all boards equally – assuming the use of E1 steppers on all boards. This should be fine for now, because currently for Dual Y or Dual Z you must have only 1 extruder. In the future these pins might be defined based on the number of extruders in-use.

To sum up, your PR would never have been merged in its original form, as it wasn't appropriate or comprehensive with relation to the current code-base. More discussion and re-working has been necessary, with more close attention paid to the issue being addressed. While I feel for your disappointment over not having your PR accepted and merged (or rejected and closed) in a timely manner, your "not interested wasting more of my time" attitude is out-of-line. We are all just helpful volunteers here, and it just happens that the squeakiest and most comprehensive issues get most of the elbow grease.

**Comment by thinkyhead** [comment]
https://github.com/MarlinFirmware/MarlinDev/pull/108 would be the most comprehensive set of changes.

**Comment by d00616** [comment]
@thinkyhead Thank you for your commitment and you clean words. The posted link was my mistake. The Github PR Mail address was found a way into my Mail addressbook. This link was for an person starting with the same first letter.

**Comment by thinkyhead** [comment]
@d00616 Someone should be merging those changes soon, or maybe I'll do it myself…. It's good to get this stuff fixed up and more consistent!
